flake.nix

```diff
@@ -0,0 +1,125 @@
+{
+  inputs = {
+    # dream2nix.url = "github:nix-community/dream2nix";
+    dream2nix.url = "github:blaggacao/dream2nix";
+  };
+
+  outputs = {
+    self,
+    dream2nix,
+    nixpkgs,
+  } @ inp: let
+    inherit (nixpkgs.legacyPackages.x86_64-linux) pkgs;
+    inherit (nixpkgs) lib;
+  in
+    (dream2nix.lib.makeFlakeOutputs {
+      systems = ["x86_64-linux"];
+      config.projectRoot = ./.;
+      config.packagesDir = "./locks";
+      source = ./.;
+      settings = [
+        {subsystemInfo.noDev = false;}
+      ];
+      packageOverrides = let
+        system = "x86_64-linux";
+        toplevel = "cardano-sdk";
+        subpackages = lib.filterAttrs (k: v: lib.hasPrefix "@cardano-sdk" k) self.packages.${system};
+        subpackagesNames = builtins.attrNames subpackages;
+        topDependencies =
+          builtins.removeAttrs
+          self.packages.${system}.${toplevel}.dependencies
+          (subpackagesNames ++ [toplevel]);
+      in
+        lib.recursiveUpdate (
+          builtins.mapAttrs (n: _: let
+            otherNames = lib.remove n subpackagesNames;
+            otherSubpackages = builtins.removeAttrs subpackages [n];
+          in {
+            # install-method.installMethod = "copy";
+            add-inputs.buildInputs = old: old ++ (lib.attrValues topDependencies) ++ [pkgs.nodejs-14_x.pkgs.yarn];
+            install-symlinks.postInstall = "ln -s $out/lib/node_modules/${n}/dist $out/dist";
+            add-type-notation.prePatch = lib.optionalString (n == "@cardano-sdk/core")
+            ''
+              substituteInPlace ./src/Cardano/util/subtractValueQuantities.ts \
+                --replace "../types"  "@cardano-sdk/core/src/Cardano/types"
+            '';
+            fix-relative-references.postPatch = let
+              references = {
+                "@cardano-sdk/blockfrost" = [ "core" "util" ];
+                "@cardano-sdk/cardano-services-client" = [ "core" "util" ];
+                "@cardano-sdk/cardano-services" = [ "core" "ogmios" "rabbitmq" "util" ];
+                "@cardano-sdk/cip2" = [ "core" "util" ];
+                "@cardano-sdk/cip30" = [ "core" ];
+                "@cardano-sdk/core" = [ "util" ];
+                "@cardano-sdk/e2e" = [ "util" "util-dev" "wallet" "cardano-services" ];
+                "@cardano-sdk/golden-test-generator" = [ "core" "util" ];
+                "@cardano-sdk/ogmios" = [ "core" ];
+                "@cardano-sdk/rabbitmq" = [ "core" "util" ];
+                "@cardano-sdk/util" = [];
+                "@cardano-sdk/util-dev" = [ "core" ];
+                "@cardano-sdk/util-rxjs" = [];
+                "@cardano-sdk/wallet" = [ "cip2" "cip30" "core" "util-rxjs" "util"];
+                "@cardano-sdk/web-extension" = [ "cip30" "core" "wallet" "util-rxjs" "util"];
+              };
+              replacer = lib.concatMapStrings (name: ''
+                --replace \
+                  '"path": "../../${name}/src"' \
```

Comment:
This should only be needed for `packages/[package]/test` projects. Are we running `tsc` for test projects? If we do I suggest omitting it to make maintenance easier

Reply:
The relative location between packages is not guaranteed, and hence `../../other` can point to nowhere.

This is compartmentalization is actually an important feature for proper package-wise build caching.

Reply:
There shouldn't be any `../../other` imports outside of test code. Test code is only included in `[package]/test/tsconfig.json` and we should only be concerned about building with `[package]/src/tsconfig.json` unless we also want to build the tests (which is not even supported, we just use `ts-node` to run them)